### DISCUSSION TITLE
device_tracker set up

### DISCUSSION THREAD

**Discussion by lmdc45** [Vaskivskyi/ha-asusrouter#582] [General]
I can see all devices properly in the 'tracked_devices" sections when I download the device diagnostic. However I cannot get the device tracking system to work
events notification are still working (the connection example you have)
I understand there is nothing to add in confugartions.yaml file as it is done only from the ui? 

I don't get any device entities added on ha.

Is there any settings I need to add to get device entities available?

I tried adding 

device_tracker:
- platform: asusrouter
   track_new_devices: true

to configurations with no success

**Comment by lmdc45** [comment]
so I realized entities are available, however need to be manually enabled for each device from the entities panel.

Is it normal and is it possible to automatically track all devices?

**Comment by Vaskivskyi** [comment]
Hello, @lmdc45,

Sorry for the late reply. Unfortunately, lately, I didn't have the possibility to check GitHub and answer issues and questions.

Yes, the `disabled` state of device trackers is the normal standard behaviour of HA. The only case in which device trackers are enabled by default - is when the device MAC is already known to HA from some other integration. Unfortunately, I cannot influence this behaviour anyhow.

**Reply by lmdc45** [reply]
Thanks for your reply. I am new to ha. 
Also I could set up your example for device reconnected event. I also added the other two events however I only receive the reconnected event

**Reply by Vaskivskyi** [reply]
In order to use HA events, please enable them in the integration configurations.

If all of them (`connected`, `disconnected` and `reconnected`) are enabled, but you are getting only some please check the following:

1. `connected` event is raised only for a device which was not tracked before by HA. If the device was already tracked but was offline, it will only raise a `reconnected` event.
2. Disconnected event is raised the same way as the `not home` state for device trackers - there is a configurable `consider home` time for each device, only after which device disconnect will be raised. So in general, the event will raise exactly when the `device_tracker` entity will change its state (even though, they work independently)

In case you are sure events should be raised, but they do not for your setup, please open an issue with as many details as possible, so we can reproduce, debug and fix the issue.
